src/client/cl_view.c

```diff
@@ -194,20 +189,15 @@ void Cl_UpdateView(void) {
 	// set area bits to mark visible leafs
 	r_view.area_bits = cl.frame.area_bits;
 
-	const cl_frame_t *frame = &cl.frames[(cl.frame.frame_num - 1) & PACKET_MASK];
-	const player_state_t *from = frame->frame_num == cl.frame.frame_num - 1 ? &frame->ps : &cl.frame.ps;
+	const player_state_t *ps = cl.delta_frame ? &cl.delta_frame->ps : &cl.frame.ps;
 
-	Cl_UpdateOrigin(from, &cl.frame.ps);
+	Cl_UpdateOrigin(ps, &cl.frame.ps);
 
-	Cl_UpdateAngles(from, &cl.frame.ps);
+	Cl_UpdateAngles(ps, &cl.frame.ps);
 
```

Comment:
This delta player state logic was always unnecessarily ugly. Just use cl.delta_frame.